components/tikv_kv/src/lib.rs

```diff
@@ -565,7 +563,7 @@ pub unsafe fn destroy_tls_engine<E: Engine>() {
 }
 
 /// Get a snapshot of `engine`.
-pub fn snapshot<E: Engine>(
+pub fn snapshot<E: Storage>(
     engine: &E,
```

Comment:
```suggestion
    storage: &S,
```
same for other places

Reply:
I'm afraid this may make most cherry picks after this PR conflict with the pass.